src/numeral_systems.rs

```diff
@@ -0,0 +1,715 @@
+                self.number,
+            ),
+            NumeralSystem::EasternArabicPersian => positional(
+                f,
+                &['۰', '۱', '۲', '۳', '۴', '۵', '۶', '۷', '۸', '۹'],
+                self.number,
+            ),
+            NumeralSystem::DevanagariNumber => positional(
+                f,
+                &['०', '१', '२', '३', '४', '५', '६', '७', '८', '९'],
+                self.number,
+            ),
+            NumeralSystem::BengaliNumber => positional(
+                f,
+                &['০', '১', '২', '৩', '৪', '৫', '৬', '৭', '৮', '৯'],
+                self.number,
+            ),
+            NumeralSystem::Symbol => {
+                symbolic(f, &['*', '†', '‡', '§', '¶', '‖'], self.number)
+            }
+        }
+    }
+}
+
+/// Formats a number using a
+/// [sign-value notation](https://en.wikipedia.org/wiki/Sign-value_notation).
+///
+/// The symbols must be specified by decreasing values.
+///
+/// The value of a stringified number is recovered by summing over the values of
+/// the symbols present.
+///
+/// Consider the situation where `[("V", 5), ("IV", 4), ("I", 1)]` are the
+/// provided symbols:
+///
+/// ```text
+/// 1 => 'I'
+/// 2 => 'II'
+/// 3 => 'III'
+/// 4 => 'IV'
+/// 5 => 'V'
+/// 6 => 'VI'
+/// 7 => 'VII'
+/// 8 => 'VIII'
+/// ```
+///
+/// This is the start of the familiar Roman numeral system.
+fn additive(
+    f: &mut Formatter<'_>,
+    symbols: &[(&str, u64)],
+    mut n: u64,
+) -> std::fmt::Result {
+    if n == 0 {
+        if let Some(&(symbol, 0)) = symbols.last() {
+            return write!(f, "{}", symbol);
+        }
+        return write!(f, "0");
+    }
+
+    for (symbol, weight) in symbols {
+        if *weight == 0 || *weight > n {
+            continue;
+        }
+        let reps = n / weight;
+        for _ in 0..reps {
+            write!(f, "{}", symbol)?
+        }
+
+        n -= weight * reps;
+    }
+    Ok(())
+}
+
+/// Formats a number using a big-endian
+/// [bijective base-_b_](https://en.wikipedia.org/wiki/Bijective_numeration)
+/// system (where _b_ is the number of provided symbols). This is similar to
+/// regular base-_b_ systems, but without a symbol for zero.
+///
+/// Consider the situation where `['A', 'B', 'C']` are the provided symbols:
+///
+/// ```text
+/// 1 => "A"
+/// 2 => "B"
+/// 3 => "C"
+/// 4 => "AA"
+/// 5 => "AB"
+/// 6 => "AC"
+/// 7 => "BA"
+/// ...
+/// ```
+///
+/// A similar system is commonly used in spreadsheet software.
+fn bijective(f: &mut Formatter<'_>, symbols: &[char], mut n: u64) -> std::fmt::Result {
+    if n == 0 {
+        return write!(f, "-");
+    }
+
+    let radix = symbols.len() as u64;
+    let mut digits = Vec::with_capacity((n.ilog(radix) + 1) as usize);
+    while n != 0 {
+        n -= 1;
+        digits.push(symbols[(n % radix) as usize]);
+        n /= radix;
+    }
+    for digit in digits.iter().rev() {
+        write!(f, "{}", digit)?
+    }
+    Ok(())
+}
```

Comment:
The formula for `size` is from Wikipedia, the rest was actually stupidly easy: Just subtract 1111...111 in the given base and apply the positional algorithm.
We probably don't need the special case for `radix = 1`, but I still added it just in case.
```suggestion
fn bijective(f: &mut Formatter<'_>, symbols: &[char], mut n: u64) -> std::fmt::Result {
    if n == 0 {
        return write!(f, "-");
    } else if let [one] = symbols {
        for _ in 0..n {
            write!(f, "{one}")?;
        }
        return Ok(());
    }

    let radix = symbols.len() as u64;
    let size = ((n + 1) * (radix - 1)).ilog(radix);
    n -= (radix.pow(size) - 1)/(radix - 1);
    // For a number of size 1, the MSD's place is the ones place, hence `- 1`.
    let mut most_significant_digit_place = radix.pow(size - 1);
    for _ in 0..size {
        let most_significant_digit = n / most_significant_digit_place;
        write!(f, "{}", symbols[most_significant_digit as usize])?;
        n -= most_significant_digit * most_significant_digit_place;
        most_significant_digit_place /= radix;
    }
    Ok(())
}
```

Reply:
> Just subtract 1111...111 in the given base and apply the positional algorithm.

With hindsight this is so obvious. Thanks!

The case for base-1 is already not treated properly in `positional` and we don't need it for now, so let's not worry about it for now.